examples/nfct_test.rs

```diff
@@ -0,0 +1,170 @@
+#![no_std]
+
+extern crate alloc;
+extern crate lang_items;
+extern crate libtock_drivers;
+
+use core::fmt::Write;
+use libtock_core::result::CommandError;
+use libtock_drivers::console::Console;
+use libtock_drivers::nfc::NfcTag;
+use libtock_drivers::nfc::RecvOp;
+use libtock_drivers::result::FlexUnwrap;
+use libtock_drivers::result::TockError;
+use libtock_drivers::timer;
+use libtock_drivers::timer::Timer;
+use libtock_drivers::timer::Timestamp;
+
+#[allow(dead_code)]
+/// Helper function to write on console the received packet.
+fn print_rx_buffer(buf: &mut [u8], amount: usize) {
+    if amount < 1 || amount > buf.len() {
+        return;
+    }
+    let mut console = Console::new();
+    write!(console, "RX:").unwrap();
+    for byte in buf.iter().take(amount - 1) {
+        write!(console, " {:02x?}", byte).unwrap();
+    }
+    writeln!(console, " {:02x?}", buf[amount - 1]).unwrap();
+    console.flush();
+}
+
+#[allow(dead_code)]
+/// Function to identify the time elapsed for a transmission request.
+fn bench_transmit(console: &mut Console, timer: &Timer, title: &str, mut buf: &mut [u8]) {
+    let amount = buf.len();
+    let start = Timestamp::<f64>::from_clock_value(timer.get_current_clock().flex_unwrap());
+    match NfcTag::transmit(&mut buf, amount) {
+        Ok(_) => (),
+        Err(_) => writeln!(Console::new(), " -- tx error!").unwrap(),
+    }
+    let end = Timestamp::<f64>::from_clock_value(timer.get_current_clock().flex_unwrap());
+    let elapsed = (end - start).ms();
+    writeln!(
+        console,
+        "{}\n{:.2} ms elapsed for {} bytes ({:.2} kbit/s)",
+        title,
+        elapsed,
+        amount,
+        (amount as f64) / elapsed * 8.
+    )
+    .unwrap();
+    console.flush();
+}
+
+#[derive(PartialEq, Eq)]
+/// enum for reserving the NFC tag state.
+enum State {
+    Enabled,
+    Disabled,
+}
```

Comment:
Are you expecting more variants, or why not just use a simple `bool is_enabled`?

Reply:
I wanted to use the match pattern with what I thought a more readable tag state.

Reply:
Not wrong per se, but for me, the additional code for the `enum` is not worth it. Maybe others disagree, but it has other readability advantages. The nesting is a bit more shallow. `Enabled` vs `Disabled` are direct opposites, so you can reuse the `bool` idea, that people are used to.